rdm/0002-data-models-terminology.md

```diff
@@ -0,0 +1,38 @@
+- Start Date: 2019-09-30
+- RFC PR: [#2](https://github.com/inveniosoftware/rfcs/pull/2)
+- Authors: Lars Holm Nielsen
+
+# Data models terminology
+
+## Summary
+
+Definition of terms used when speaking about records and data models.
+
+## Motivation
+
+The purpose of this RFC is to disambiguate terms like records, record type, schema and data model that are often used interchangeably in the context of Invenio.
+
+## Detailed design
+
+The central part of InvenioRDM is the primary data model for storing a bibliographic record type describing the digital resources (datasets, software, articles, ...). In addition to the primary data model, InvenioRDM can be extended with one or more secondary data models for storing authority record types (grants, licenses, ...).
+
+You may often see the words like records, record type, schema and data model used interchangeably in the context of Invenio, therefore it’s useful to distinguish what is meant with each term:
+
+- **Record:** a JSON document describing an object.
+- **Record type:** a category of JSON documents describing a specific type of objects (examples: grants, licenses, authors, research outputs).
+- **Record schema:** a JSONSchema describing the JSON structure of a record belonging to a specific record type. A record type may have one or more record schemas (examples: authors schema v1.0, authors schema v1.1, ...).
+- **Deposit record:** a temporary record (e.g. copy of an existing record) used for supporting long running edit sessions that allow users to save their partial changes to a record. A deposit record belongs to the same record type as the source record.
```

Comment:
I am used to it now, but I remember the difficulty I had in grasping this concept because of `invenio-deposit`'s implementation. In `invenio-deposit`, a `Deposit` has both the "draft" feature outlined here **AND** the file association feature. Was something a "deposit" because it had files (files are deposited) or because it was a draft? There might be some disambiguation needed here.

Reply:
> A deposit record belongs to the same record type as the source record.

Shall we say that there is only one 'record type' for deposits? Authorities do not have deposits.